src/portable/synopsys/dwc2/dcd_dwc2.c

```diff
@@ -630,6 +630,37 @@ void dcd_sof_enable(uint8_t rhport, bool en)
 /* DCD Endpoint port
  *------------------------------------------------------------------*/
 
+#if TU_CHECK_MCU(OPT_MCU_ESP32S2, OPT_MCU_ESP32S3, OPT_MCU_ESP32P4)
+
+// Check if IN/OUT endpoint is avaliable before opening it
+static bool dcd_ep_available(uint8_t rhport, uint8_t dir) 
+{
+    // Verify that we have a vacant EP
+    if ((dwc_ep_config[rhport].in_ep + dwc_ep_config[rhport].out_ep) > (dwc_ep_config[rhport].ep_max_count - 1)) {
+        TU_LOG(1, "Trying to open an endpoint, but max number of endpoints: %d already opened on this target \r\n", dwc_ep_config[rhport].ep_max_count);
+        return false;
+    }
+    // Get the ep_count amount at the moment as a temporal variable and update it
+    uint8_t new_ep_count = (dir) ? dwc_ep_config[rhport].in_ep : dwc_ep_config[rhport].out_ep;
```

Comment:
Seems like `new_ep_count++` is missing. No?

Reply:
Right, I was doing some HW testing and forgot to put it back.